openlibrary/plugins/openlibrary/js/edit.js

```diff
@@ -439,17 +439,33 @@ export function initEditLinks() {
         },
         validate: function(data) {
             const i18nStrings = JSON.parse(document.querySelector('#link-errors').dataset.i18n);
+            const url = data.url.trim();
+
+            function isValidURL(url) {
```

Comment:
Toss this function at the bottom of the file; it's sufficiently general purpose, and will make this code block a little clearer if it's not here.